### PR TITLE
Script Changes

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Change script to ask for liferay folder instead of assuming. Also move python script to liferay folder in case it was not moved manually

### CODE REVIEW

File: gen-idea.sh [hunk 1]
```diff
@@ -1,40 +1,38 @@
 #!/bin/sh
 
 # Check if current directory is the liferay-portal project folder
-PORTAL=$PWD
-PORTAL_DEFAULT=$HOME/projects/liferay-portal
+PORTAL=$HOME/projects/liferay-portal
 
 if [ ! -f $PORTAL/gradlew ]; then
-    echo "$PORTAL is not a valid liferay project folder, falling back to $PORTAL_DEFAULT..."
+	echo "$PORTAL is not a valid liferay project folder..."
```
Comment (andreldm) [inline gen-idea.sh]: Seriously, tabs even here?

**Comment by andreldm** [comment]
These changes break the way I use this script, as an alias, no need to move /ideactl.py around or type long paths. So NOT-LGTM.

**Comment by rodrigopaulino** [comment]
Don't know about the ideactl.py (which is not working btw), but the attempt here is to make the script reusable for anyone with other set of directories. If it doesn't work for your alias solution, you should export the $PORTAL env var instead. That will probably do it. If after that it still doesn't suit your desires, well... ¯\_(ツ)_/¯

**Comment by rodrigopaulino** [comment]
Yeah. My proposed solution won't work for your case either since the $PORTAL is overwritten anyway at the beginning of the script. Let me fix that for you. I'll set the initial $PORTAL value to $PWD. Does this work for you?
Regarding the .py file being moved to Liferay's root folder, @pedroqueiroz told me it was necessary to manually move that file to the Liferay project folder before running it (don't know why, didn't try to read the ideactl.py) and that's what originated this pull request. If it's not necessary to move the .py, this pull is pointless. Although, I still think it's better UX to ask for the full path of the Liferay project instead of assuming a couple of directories and that's it.

File: gen-idea.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 # Check if current directory is the liferay-portal project folder
-PORTAL=$HOME/projects/liferay-portal
+PORTAL=$PWD
```
Comment (rodrigopaulino) [inline gen-idea.sh]: @andreldm, now you can call your alias from the liferay project directory or add this command to your alias: "cd $HOME/projects/liferay-portal".

**Comment by andreldm** [comment]
The rationale is to call that script on whatever directory the terminal is on, so instead of the user telling where is the portal workspace, he just needs to open a terminal there (via Finder or taking advantage of shell auto completion). It's much better for users with multiple portal workspaces (ce, dxp, 6.2...), where a single envvar wouldn't work.

AFAIR that python script is necessary and it requires the CWD to be the portal workspace. The ultimate goal is to have one single script, either python or bash, easily installable with `wget -o /usr/bin/local/gen-idea.sh http://somewhere/gen-idea.sh`.

And finally, UX-wise it's better to avoid asking the user, the script should just work out of box.

**Comment by rodrigopaulino** [comment]
Hey, @andreldm.

I agree with the following statements:
> It's much better for users with multiple portal workspaces (ce, dxp, 6.2...), where a single envvar wouldn't work

and

> the script should just work out of box.

But clearly the script wasn't doing that. So instead of exiting the script, the user has two options now: run it from the liferay project folder, or tell where it is located. Much better than the previous second option.

Also, here

> no need to move /ideactl.py around

and

> AFAIR that python script is necessary and it requires the CWD to be the portal workspace

How would the script work out of the box if I don't move the python script to where it should be?

**Comment by rodrigopaulino** [comment]
I see now what you mean about the /ideactl.py
Just tested here. It doesn't matter where the python script is, but the terminal should be in the liferay project.
Yeah. This pull breaks that. I'll close it. 
Thanks.

Ps: still think it's better asking the user for the workspace path tho ;)

**Comment by andreldm** [comment]
> So instead of exiting the script, the user has two options now: run it from the liferay project folder, or tell where it is located. Much better than the previous second option.

No need to edit the script, just `$ cd path/to/portal/workspace && gen-idea.sh`.

After all, it's same as `$ gen-idea.sh path/to/portal/workspace` or `$ cd path/to/portal/workspace && gen-idea.sh $(pwd)`, I prefer the first one because portal developers usually have a running terminal emulator pointing to portal workspace.

Therefore, I think it's nice to accept the path to portal workspace as first argument, if missing assume pwd. In both cases, do not ask if the path is invalid, because `read` has no auto completion and it won't remember paths.

> How would the script work out of the box if I don't move the python script to where it should be?

Because the script is called from an absolute path (alias), so it's possible to call the python script from an absolute path assuming they are the same location:
https://github.com/pedroqueiroz/liferay-portal-gen-idea/blob/master/gen-idea.sh#L40

Not sure this would work if both are in `/usr/local/bin`.

**Comment by rodrigopaulino** [comment]
Agreed!
I just used "read" because I don't know better function! Haha

**Pull request closed without merging**